src/routes/api/user/index.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  FastifyPluginAsyncTypebox,
+  Type
+} from '@fastify/type-provider-typebox'
+import { Auth } from '../../../schemas/auth.js'
+import { UpdateCredentialsSchema } from '../../../schemas/users.js'
+
+const plugin: FastifyPluginAsyncTypebox = async (fastify) => {
+  fastify.put(
```

Comment:
I think we should use a rate limiter to mitigate brute force attack, there is an example here: https://github.com/fastify/demo/blob/4470023655ea4e68de65d9b4e78a6f03c56ab49a/src/app.ts#L74